Make the robot keep track of the full last drink so the display can too

audiodude opened the pull request:
Includes some reformatting of the display HTML to make everything fit. Also
changes the format of drink-queue.json.

audiodude committed:
Make the robot keep track of the full last drink so the display can as well.

Includes some reformatting of the display HTML to make everything fit. Also
changes the format of drink-queue.json.

audiodude committed:
Only show 'sitting around' message when there are no drinks.

funrollloops commented on poll_appengine.py:
```diff
@@ -64,6 +65,9 @@ def __init__(self, base_url, poll_frequency_secs, controller):
     self.queuefile = 'remote_queue.txt'
     self.jsonqueuefile = 'static/order-queue.json'
+    self.queue_for_display = {}
+    self.finished_drink = None
```
Default to an empty dictionary here

audiodude replied:
Done

funrollloops commented on poll_appengine.py:
```diff
@@ -106,26 +110,25 @@ def post(self, url, **kwargs):
           drink_id = json_recipe['id']
-          if drink_id == last_drink_id:
-            print "Refusing to remake order %s" % last_drink_id
+          if self.finished_drink and drink_id == self.finished_drink['id']:
```
Use finished_drink.get here, the possible race here makes me nervous

audiodude replied:
Done.

funrollloops commented on poll_appengine.py:
```diff
@@ -136,20 +139,28 @@ def run(self):
+      'queue': queue or []
+    }
+
   def write(self, queue):
```
It's kind of wonky that the queue argument is ignored here -- maybe make queue_for_display a local variable in this function, and inline prepare_queue_for_display here?

audiodude replied:
Got rid of self.queue_for_display as a member variable and inlined it. Good idea!

funrollloops commented on static/display.html:
```diff
@@ -9,56 +9,64 @@
+	      <div class="expl">
+	        * If you've ordered and don't see your name, talk to<br>
+          a specialist to get your drink approved.
+	      </div>
```
indentation is a little wonky, maybe you're using tabs?

audiodude replied:
Fixed, thanks!

audiodude committed:
Merge branch 'master' of https://github.com/audiodude/nebree8

audiodude committed:
Merge branch 'master' of https://github.com/nebree8/nebree8

audiodude committed:
Update poll_appengine to clear current_drink when queue is empty.

Add progress percent to current_drink for display.

Change some display stying and fix conditionals for default empty object.

audiodude reviewed (commented):
I made additional changes, including fixing bugs that were present when there was no queue, or no current_drink (like clearing out the current drink so the display doesn't display it when the queue is empty).

I also added updating the progress_percent on the syncer's local current_drink copy, so that can be reflected in the display.

http://i.imgur.com/f3XOia3.png

PTAL, thanks.
